src/encoder.rs

```diff
@@ -570,11 +570,28 @@ pub struct SegmentationState {
   pub last_active_segid: u8,
   pub features: [[bool; SegLvl::SEG_LVL_MAX as usize]; 8],
   pub data: [[i16; SegLvl::SEG_LVL_MAX as usize]; 8],
-  pub segment_map: [usize; 8],
+  pub threshold: [DistortionScale; 7],
   pub min_segment: u8,
   pub max_segment: u8,
 }
 
+impl SegmentationState {
+  pub fn update_threshold(&mut self, base_q_idx: u8, bit_depth: usize) {
+    use itertools::Itertools;
+    let base_ac_q = ac_q(base_q_idx, 0, bit_depth) as u64;
+    let threshold = (0..=self.max_segment as usize)
+      .map(|i| self.data[i][SegLvl::SEG_LVL_ALT_Q as usize] as i8)
```

Comment:
maybe the two maps could be collapsed in one and you could iterate directly over a slice of data.